crates/full-node/sov-db/src/flat_db.rs

```diff
@@ -20,26 +22,71 @@ pub struct FlatStateDb {
     pub(crate) user: VersionedDB<NomtStateValues<UserNamespace>>,
     pub(crate) kernel: VersionedDB<NomtStateValues<KernelNamespace>>,
     pub(crate) other: Arc<rockbound::DB>,
+    #[allow(dead_code)]
+    // We don't technically need to store the archival db here - it's only accessed through the user/kernel versioned DB wrappers.
+    // We keep it here so that the internal structure is more legible by glancing at this struct. Note that unless the user has set the config to separate out the archival db,
+    // this will point to the same rockbound::DB as the `other` field.
+    pub(crate) archival: Arc<rockbound::DB>,
+}
+
+pub struct FlatDbCommitData {
+    pub user_archival: SchemaBatch,
+    pub kernel_archival: SchemaBatch,
```

Comment:
Stupid question, but why live data is in a single schema batch, both kernel and user, but archival is split into 2?

Reply:
Not a dumb question at all. This was an artifact of the old API setup. Tweaked rockbound to let us unify them.